cpp/src/arrow/csv/reader.cc

```diff
@@ -1139,8 +1177,9 @@ class CSVRowCounter : public ReaderMixin,
   }
 
   Future<int64_t> DoCount(const std::shared_ptr<CSVRowCounter>& self) {
-    // We must return a value instead of Status/Future<> to work with MakeMappedGenerator,
-    // and we must use a type with a valid end value to work with IterationEnd.
+    // We must return a value instead of Status/Future<> to work with
+    // MakeMappedGenerator, and we must use a type with a valid end value to work with
```

Comment:
```suggestion
    // count_cb must return a value instead of Status/Future<> to work with
    // MakeMappedGenerator, and it must use a type with a valid end value to work with
```